robotPositioner/robotPositioner.h

```diff
@@ -0,0 +1,382 @@
+    unsigned int m_vicon_udp_client_port;                     // udp streaming port number
+    std::string m_vicon_capture_control_ip_address;           // vicon capture control ip address
+    std::string m_vicon_capture_control_executable_path;      // vicon capture control program executable path
+    unsigned int m_capture_control_port;                      // vicon capture control port number
+    double m_robot_r;                                         // robot's wheel radius
+    double m_robot_D;                                         // robots' axis length
+    double m_allowed_heading_error;                           // the amount of error allowed in the final heading
+    double m_stopping_distance;                               // if the robot's distance from goal < stopping dist, robot stops
+    std::string m_video_device;                               // video device path (usually /dev/device0 )
+    double m_k1;                                              // curveness of the path to the goal
+    double m_k2;                                              // speed of turning on the curves
+    double m_alpha;                                           // causes more sharply peaked curves 
+    double m_beta;                                            // causes to drop velocity if 'k'(curveness) increases
+    double m_max_velocity;                                    // max velocity
+
+    /**
+    * sets up the vicon system so it listens to the broadcast 
+    */
+    void setupVicon() {
+
+        // setup vicon and capture control
+        m_vicon = new UDPClient<ObjectData>(m_vicon_udp_client_port);
+        m_viconCaptureControl = new CaptureControl( 
+            m_vicon_capture_control_ip_address,
+            m_capture_control_port,
+            m_vicon_capture_control_executable_path
+        );
+
+        // keep trying to get stream of data
+        while(m_vicon->getNumObjects() == 0) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            std::cout << "Waiting for object" << std::endl;
+        }
+
+        // start recording 
+        if(!m_viconCaptureControl->startRecording("test1")) {
+            std::cout << "failed to start recording" << std::endl;
+        } else {
+            std::cout << "vicon finished setting up" << std::endl;
+        }
+    }
+
+
+    // sets up the 360 SeeCam
+    void setupCamera360() {
+        //const std::string device = "/dev/video0";
+        const std::string device = m_video_device;
+
+        m_cam = new See3CAM_CU40(device, See3CAM_CU40::Resolution::_1280x720);
+        const cv::Size unwrapSize(180,50); // calculate camera input dimensions
+
+        const cv::Mat bubblescopeMask = See3CAM_CU40::createBubblescopeMask(m_cam->getSuperPixelSize());
+        m_cam->autoExposure(bubblescopeMask);
+        m_unwrapper = m_cam->createUnwrapper(unwrapSize);
+    }
+
+    // gets vicon data
+    double getViconData() {
```

Comment:
This function should be ``void`` because it doesn't return anything. I think the name should probably be something like ``updateViconData()`` because otherwise it sounds like it should return something.